### PR TITLE
[Form] add `choice_translation_domain` option to date types

### PR DESCRIPTION
**Pull request by HeahDude**
| Q | A |
| --- | --- |
| Doc fix? | yes |
| New docs? | no |
| Applies to | 2.8+ |
| Fixed tickets | partially #5934 |

### CODE REVIEW

File: reference/forms/types/options/datetime_choice_translation_domain.rst.inc [hunk 1]
```diff
@@ -0,0 +1,15 @@
+choice_translation_domain
```
Comment (xabbuh) [inline reference/forms/types/options/datetime_choice_translation_domain.rst.inc]: Shouldn't we reuse the already existing file from the choice type?

Comment (HeahDude) [inline reference/forms/types/options/datetime_choice_translation_domain.rst.inc]: I would have. Unfortunately there is a conflict between added version : 2.7 for choice types but 2.8 for date types.
Do you know a way to override only the note in a block like this ?

Comment (xabbuh) [inline reference/forms/types/options/datetime_choice_translation_domain.rst.inc]: Unfortunately no. But maybe we can update the description of the existing option with what you used here (imo it's more clear than what we already have).

Comment (HeahDude) [inline reference/forms/types/options/datetime_choice_translation_domain.rst.inc]: I don't understand, I've just copy-pasted the current description and changed the version added.

File: reference/forms/types/time.rst [hunk 2]
```diff
@@ -15,7 +15,8 @@ stored as a ``DateTime`` object, a string, a timestamp or an array.
 +----------------------+-----------------------------------------------------------------------------+
 | Rendered as          | can be various tags (see below)                                             |
 +----------------------+-----------------------------------------------------------------------------+
-| Options              | - `placeholder`_                                                            |
+| Options              | - `choice_translation_domain`_                                              |
+|                      | - `placeholder`_                                                            |
```
Comment (HeahDude) [inline reference/forms/types/time.rst]: Do you prefer to merge that one first or to reorder in 2.7 with a rebase here ?

**Comment by HeahDude** [comment]
This one should be ready to be merged.

**Comment by xabbuh** [comment]
👍 I like it.

**Comment by WouterJ** [comment]
Thanks @HeahDude!

I've created another include file with the duplicated body and included that file in both choice_translation_domain files in 9448b1b . That solves the duplication problem.

**Comment by HeahDude** [comment]
@WouterJ thanks, great idea! So we can only use that file in 3.x getting rid of version added notes, right?

**Comment by WouterJ** [comment]
Yes, indeed.

**Pull request merged**